tsdb/store.go

```diff
@@ -1900,136 +1914,24 @@ func (s *Store) TagValues(auth query.Authorizer, shardIDs []uint64, cond influxq
 		sort.Sort(tagValuesSlice(allResults))
 	}
 
-	// The next stage is to merge the tagValue results for each shard's measurements.
-	var i, j int
-	// Used as a temporary buffer in mergeTagValues. There can be at most len(shards)
-	// instances of tagValues for a given measurement.
-	idxBuf := make([][2]int, 0, len(is.Indexes))
-	for i < len(allResults) {
-		// Gather all occurrences of the same measurement for merging.
-		for j+1 < len(allResults) && bytes.Equal(allResults[j+1].name, allResults[i].name) {
-			j++
-		}
-
-		// An invariant is that there can't be more than n instances of tag
-		// key value pairs for a given measurement, where n is the number of
-		// shards.
-		if got, exp := j-i+1, len(is.Indexes); got > exp {
-			return nil, fmt.Errorf("unexpected results returned engine. Got %d measurement sets for %d shards", got, exp)
-		}
-
-		nextResult := mergeTagValues(idxBuf, allResults[i:j+1]...)
-		i = j + 1
+	for _, r := range allResults {
```

Comment:
Probably this used to be needed before iterators over IndexSet? Now the merge is done within the iterator.